probe-rs/targets/esp32.yaml

```diff
@@ -7,38 +7,51 @@ variants:
     cores:
       - name: main
         type: riscv
-        core_access_options:
-          !Riscv {}
-    memory_map:
+        core_access_options: !Riscv {}
+    memory_map: # From ESP32-C3 Technical Reference Manual, Table 3.2/3.3 Internal/External Memory Address Mapping
       - !Nvm
-          range:
-            start: 0
-            end: 67108864
-          is_boot_memory: true
-          cores:
-            - main
+        range:
+          start: 0x0
+          end: 0x04000000
+        is_boot_memory: true
+        cores:
+          - main
       - !Ram
-          range:
-            start: 1077411840
-            end: 1077805056
-          is_boot_memory: false
-          cores:
-            - main
+        range: # 384 Kb SRAM on Instruction Bus
+          start: 0x40380000
+          end: 0x403e0000
+        is_boot_memory: false
+        cores:
+          - main
       - !Ram
-          range:
-            start: 1070071808
-            end: 1070137344
-          is_boot_memory: false
-          cores:
-            - main
+        range: # 384 Kb SRAM on Data Bus
+          start: 0x3fc80000
+          end: 0x3fc90000
+        is_boot_memory: false
+        cores:
+          - main
+      - !Nvm
+        range: # External Flash on Instruction Bus (Read Only)
```

Comment:
Is this memory present in all configurations? Sounds board specific to me?

Reply:
I suspect "board specific" is accurate.  @MabezDev is probably the best to clarify this question.

Reply:
If this is board specific then it should go in a board specific config. We never built a mechnaism for those tho. We only ever duplicated them with a specific name.

Reply:
It's chip-specific, but not board-specific. The esp32c3 will be able to address these regions regardless of the chip size (an error will be thrown if you try and access FLASH space that doesn't exist of course). So I think this is okay to keep here?

Reply:
This seems OK to me, considering the current way memory maps work.  @Yatekii ... I will leave it to you to advise on how we proceed. Without this additional region in the memory map, it is not possible to read RTT channel names.